oprf/group/group.go

```diff
@@ -0,0 +1,117 @@
+package group
+
+import (
+	"hash"
+	"math/big"
+)
+
+// CiphersuiteID is
+type CiphersuiteID int
+
+const (
+	// P256 is the group generated by P-256 elliptic curve.
+	P256 CiphersuiteID = iota
+)
+
+// Ciphersuite corresponds to the OPRF ciphersuite that is chosen. The
+
+// Ciphersuite corresponds to the OPRF ciphersuite that is chosen. The
+// Ciphersuite object determines the prime-order group (pog) that is used for
+// performing the OPRF operations, along with the different hash function
+// definitions.
+// Should be created using New, using a CiphersuiteID constant.
+type Ciphersuite struct {
+	// name of the ciphersuite
+	name string
+
+	// context
+	ctx string
```

Comment:
Why is `context` part of the Ciphersuite struct?

Reply:
The context is defined at the start of the protocol by the SetUp function, so I was thinking that it will be good to set it globally here, so it can be reached by the several functions that use it.